dpnp/backend/src/queue_sycl.hpp

```diff
@@ -69,120 +60,95 @@ namespace mkl_rng = oneapi::mkl::rng;
-    backend_sycl &operator=(const backend_sycl &) = delete;
-
-    /**
-     * Initialize @ref queue
-     */
-    static void backend_sycl_queue_init(
-        QueueOptions selector = QueueOptions::CPU_SELECTOR);
-
-    /**
-     * Return True if current @ref queue is related to cpu device
-     */
-    static bool backend_sycl_is_cpu();
-
-    /**
-     * Initialize @ref rng_engine and @ref rng_mcg59_engine
-     */
-    static void backend_sycl_rng_engine_init(size_t seed = 1);
-
-    /**
-     * Return the @ref queue to the user
-     */
-    static sycl::queue &get_queue()
+    template <typename SeedT>
+    void set_rng_engines_seed(const SeedT &seed)
     {
-#if defined(DPNP_LOCAL_QUEUE)
-        if (!queue) {
-            backend_sycl_queue_init();
+        auto rng_eng_mt19937 =
+            std::make_shared<mkl_rng::mt19937>(*queue_ptr, seed);
+        if (!rng_eng_mt19937) {
+            throw std::runtime_error(
+                "Could not create MT19937 engine with given seed");
         }
-
-        return *queue;
-#else
-        // temporal solution. Started from Sept-2020
-        DPCTLSyclQueueRef DPCtrl_queue = DPCTLQueueMgr_GetCurrentQueue();
-        if (DPCtrl_queue == nullptr) {
-            std::string reason =
-                (DPCTLQueueMgr_GetQueueStackSize() == static_cast<size_t>(-1))
-                    ? ": the queue stack is empty, probably no device is "
-                      "available."
-                    : ".";
+        auto rng_eng_mcg59 = std::make_shared<mkl_rng::mcg59>(*queue_ptr, seed);
+        if (!rng_eng_mcg59) {
             throw std::runtime_error(
-                "Failed to create a copy of SYCL queue with default device" +
-                reason);
+                "Could not create MCG59 engine with given seed");
         }
-        return *(reinterpret_cast<sycl::queue *>(DPCtrl_queue));
-#endif
+
+        rng_mt19937_engine_ptr.swap(rng_eng_mt19937);
+        rng_mcg59_engine_ptr.swap(rng_eng_mcg59);
     }
 
-    /**
-     * Return the @ref rng_engine to the user
-     */
-    static mkl_rng::mt19937 &get_rng_engine()
+    bool backend_sycl_is_cpu() const
     {
-        if (!rng_engine) {
-            backend_sycl_rng_engine_init();
-        }
-        return *rng_engine;
+        const sycl::queue &q = *queue_ptr;
+        return q.get_device().is_cpu();
     }
 
-    /**
-     * Return the @ref rng_mcg59_engine to the user
-     */
-    static mkl_rng::mcg59 &get_rng_mcg59_engine()
+private:
+    backend_sycl()
+        : queue_ptr{}, rng_mt19937_engine_ptr{}, rng_mcg59_engine_ptr{}
     {
-        if (!rng_engine) {
-            backend_sycl_rng_engine_init();
+        const sycl::property_list &prop =
+            (is_verbose_mode())
+                ? sycl::property_list{sycl::property::queue::enable_profiling()}
+                : sycl::property_list{};
+        queue_ptr =
+            std::make_shared<sycl::queue>(sycl::default_selector_v, prop);
```

Comment:
Why do you need to call `make_shared`? `sycl::queue` is already a shared pointer.

Reply:
Changed `backend_sycl` class to store class instances instead of storing shared pointers to them.